text/src/autogluon/text/text_prediction/predictor/predictor.py

```diff
@@ -0,0 +1,407 @@
+                       f' `predictor = TextPredictor.load({self._path})`')
+        self.save(self._path)
+        return self
+
+    def evaluate(self, data, metrics=None):
+        """ Report the predictive performance evaluated for a given dataset.
+
+        Parameters
+        ----------
+        data : str or `pandas.DataFrame`
+            This Dataset must also contain the label-column with the same column-name as specified during `fit()`.
+            If str is passed, `valid_data` will be loaded using the str value as the file path.
+        metrics : str or List[str] or None
+            Name of metric or a list of names of metrics to report.
+            If it is not given, we will return the score of the stored eval_metric.
+
+        Returns
+        -------
+        ret : a single number or a dict of metric --> metric scores
+            Output
+        """
+        return self._model.evaluate(data, metrics=metrics)
+
+    def predict(self, dataset, as_pandas=True):
+        """
+
+        Parameters
+        ----------
+        dataset
+            The input dataset.
+        as_pandas
+            Whether the output will be converted to a pandas dataframe
+
+        Returns
+        -------
+        output
+            Array of predictions. One element corresponds to the prediction value of one
+        as_pandas
+            Whether to convert the output to a pandas dataframe
+
+        """
+        assert self._model is not None, 'Model does not seem to have been constructed. Have you called fit(), or load()?'
+        output = self._model.predict(dataset)
+        if as_pandas:
+            output = pd.DataFrame({self.label: output})[self.label]
+        return output
+
+    def predict_proba(self, dataset, as_pandas=True):
+        """Predict the probability from the input
+
+        Parameters
+        ----------
+        dataset
+            The dataset to use. It can either be a pandas dataframe or the path to the pandas dataframe.
+        as_pandas
+            Whether to convert the output to pandas dataframe.
+
+        Returns
+        -------
+        output
+            The output matrix contains the probability. It will have shape (#samples, #choices).
+        """
+        assert self._model is not None,\
+            'Model does not seem to have been constructed. ' \
+            'Have you called fit(), or load()?'
+        output = self._model.predict_proba(dataset)
+        if as_pandas:
+            output = pd.DataFrame(output)
+        return output
+
+    def extract_embedding(self, dataset):
+        """Extract the feature from the neural network
+
+        Parameters
+        ----------
+        dataset
+            The dataset to extract the embedding. It can either be a pandas dataframe or
+            the path to the pandas dataframe.
+
+        Returns
+        -------
+        embeddings
+            The output will have shape (#sample, #embedding dimension)
```

Comment:
"The output will have shape (#sample, #embedding dimension)"
->
"The output will be numpy array of shape (len(dataset), D), where the embedding dimension D is determined by the neural network's architecture."